plugins/services/src/js/pages/task-details/TaskFileViewer.js

```diff
@@ -13,7 +13,22 @@ import SearchLog from "../../components/SearchLog";
 import TaskDirectory from "../../structs/TaskDirectory";
 import TaskDirectoryActions from "../../events/TaskDirectoryActions";
 
-class TaskFileViewer extends React.Component {
+export function getNewRoute(routePath, params, path) {
```

Comment:
On this whole function: could we move this into something like `UrlUtil.normalizeFilePath` or `RouterUtil.buildFilePath`? `getNewRoute` sounds too generic, also I'd love to see some JSDoc comments for new functions we write.

Reply:
Is that a common thing for us to do JSDocs?

Reply:
Some methods have, some have not… i dont see a bad thing having them but they can help a lot.